### PR TITLE
Al/appeals 33700 - Update backend to determine no POA versus not listed POA

### PR DESCRIPTION
**Pull request by almorbah**
Resolves [APPEALS-33700](https://jira.devops.va.gov/browse/APPEALS-33700)

# Description
In this ticket, we are updating the backend to nave a `not listed name` of a Power Of Attorney that correlates to the claimant. Also, this information is returned to the front end when viewing the individual decision review screen.

## Acceptance Criteria
- [X] Code compiles correctly

## Testing Plan

- rails db:migrate RAILS_ENV=development
- login as a user with Intake authorization with vha business line e.g Acbauervvhah
- Navigate to http://localhost:3000/intake
- selected a High-level review or a supplemental claim 
Continue to search

- input a veteran Id and search
SELECT
- Was this form submitted through VA.gov? yes
- Select a Receipt Date 
- What is the Benefit Type? Veterans Health Administration
- Was an informal conference requested? Yes
- is the claimant someone other than the Veteran? Yes
- Select Claimant not listed
- OPT-IN from SOC/SSOC box on the from? Yes (Soc/SSoc Opt-in)
Continue to the next step

- Select Relationship to the Veteran
- Can be any of the following (Child, Other, Healthcare provider, Attorney)
- Claimant’s or Representative name
- Name not listed 
- Add issue 
- Establish a claim 

- Once successful Go to disposition page  e.g (http://localhost:3000/decision_reviews/vha/tasks/15784)
- under Appellant's Power of Attorney section
- No Known POA.
- inspect section with React Components
- check Task object
- `has_poa: true`
- ```{
  "representative_type": "not_listed",
  ...
}```

https://github.com/department-of-veterans-affairs/caseflow/assets/149511814/f7392c94-7215-4ea0-8f44-6e61013c88ea

- [ ] For feature branches merging into master: Was this deployed to UAT?

# Frontend
## User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER

 ---|---

## Storybook Story
*For Frontend (Presentation) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

# Backend
## Database Changes
*Only for Schema Changes*

* [X] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [X] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [X] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [X] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] For queries using raw sql was an explain plan run by System Team
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [X] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [X] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

## Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
  * [ ] Check that calls are wrapped in MetricService record block
* [ ] Check that all configuration is coming from ENV variables
  * [ ] Listed all new ENV variables in description
  * [ ] Worked with or notified System Team that new ENV variables need to be set
* [ ] Update Fakes
* [ ] For feature branches: Was this tested in Caseflow UAT

# Best practices
## Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Test Coverage
Did you include any test coverage for your code? Check below:
- [X] RSpec
- [ ] Jest
- [ ] Other

### Code Climate
Your code does not add any new code climate offenses? If so why?
- [ ] No new code climate issues added

### Monitoring
- [ ] Are performance metrics (e.g., response time, throughput) being tracked?
- [ ] Are key application components monitored (e.g., database, cache, queues)?
- [ ] Is there a system in place for setting up alerts based on performance thresholds?

### Logging
- [ ] Are logs being produced at appropriate log levels (debug, info, warn, error, fatal)?
- [ ] Are logs structured (e.g., using log tags) for easier querying and analysis?
- [ ] Are sensitive data (e.g., passwords, tokens) redacted or omitted from logs?
- [ ] Is log retention and rotation configured correctly?
- [ ] Are logs being forwarded to a centralized logging system if needed?

### Auditing
- [ ] Are user actions being logged for audit purposes?
- [ ] Are changes to critical data being tracked ?
- [ ] Are logs being securely stored and protected from tampering or exposing protected data?

### Error Handling
- [ ] Are errors being caught and handled gracefully?
- [ ] Are appropriate error messages being displayed to users?
- [ ] Are critical errors being reported to an error tracking system (e.g., Sentry, ELK)?
- [ ] Are unhandled exceptions being caught at the application level ?

### Exception Handling
- [ ] Are custom exceptions defined and used where appropriate?
- [ ] Is exception handling consistent throughout the codebase?
- [ ] Are exceptions logged with relevant context and stack trace information?
- [ ] Are exceptions being grouped and categorized for easier analysis and resolution?

### CODE REVIEW

File: app/models/decision_review_intake.rb [hunk 1]
```diff
@@ -49,12 +49,11 @@ def create_claimant!
       payee_code: (need_payee_code? ? request_params[:payee_code] : nil)
     )
     if claimant&.unrecognized_claimant?
-      unless request_params[:benefit_type] == "vha"
-        claimant.save_unrecognized_details!(
-          request_params[:unlisted_claimant],
-          request_params[:poa]
-        )
-      end
+      claimant.save_unrecognized_details!(
```
Comment (brandondorner) [inline app/models/decision_review_intake.rb:52]: Nice, I was gonna make a comment how we'd have to clean up a temporary solution from Prajwal's PR but you already did it

**Review by brandondorner — APPROVED** [review]
Very cool man
The only concern I had originally looking at this was the 
`  has_one :unrecognized_appellant` 
line for the `NotListedPowerOfAttorney` class

But after looking at it some more I think this implementation works well and I don't have any issues with it

Nice job dude

File: app/models/decision_review_intake.rb (hunk 1 above)
Comment (almorbah) [reply app/models/decision_review_intake.rb:52]: It worked out pretty well

**Review by seancva — DISMISSED** [review]
LGTM, nice work!

**Review by pamatyatake2 — COMMENTED** [review]
lgtm

**Review by pamatyatake2 — APPROVED** [review]

**Review by brandondorner — DISMISSED** [review]

**Review by brandondorner — APPROVED** [review]

**Pull request merged**